.github/workflows/qa.yml

```diff
@@ -0,0 +1,47 @@
+name: Run QA Framework tests on Github Actions
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```

Comment:
I recommend having a `workflow_dispatch:` action as well, as this allows you to rebuild via a UI click.